### DISCUSSION TITLE
Can't select data as admin

### DISCUSSION THREAD

**Discussion by mjdNjhNJ** [pocketbase/pocketbase#4784] [Q&A] [answered]
I have authenticated as admin and I'm trying to select data. Code consists of snippets from documentation.

```
import PocketBase from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

const authData = await pb.admins.authWithPassword('admin@admin.admin', '1234567890');

// after the above you can also access the auth data from the authStore
console.log(pb.authStore.isValid);
console.log(pb.authStore.token);
console.log(pb.authStore.model.id);

// "logout" the last authenticated account
pb.authStore.clear();

// fetch a paginated records list
const resultList = await pb.collection('players').getList(1, 50, {
    filter: 'created >= "2022-01-01 00:00:00" && someField1 != someField2',
});

// you can also fetch all records at once via getFullList
const records = await pb.collection('players').getFullList({
    sort: '-created',
});

// or fetch only the first record that matches the specified filter
const record = await pb.collection('players').getFirstListItem('someField="test"', {
    expand: 'relField1,relField2.subRelField',
});
```

I'm left with error 403: "Only admins can perform this action.". I thought it's because of a logout line but when I move `pb.authStore.clear();` at the end I get 400: "Something went wrong while processing your request.". I think this shouldn't be the case when using the snippets from documentation. What could be wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
The 403 error is expected if you are calling `pb.authStore.clear()` before your actions because it is practically a "logout" operation.

The 400 error is most likely because you have an invalid or missing `filter` field (Note that in the code sample `someField1`, `someField2`, etc. are just an example and they need to be replaced with your actual fields). For more detailed errors you can check the logs in the _Admin UI > Logs_ and/or start the executable with `--dev`.

If the above doesn't help, then please elaborate more clearly how you are testing and export your collections as JSON (_Admin UI > Settings > Export collections_) and I'll try to investigate it further.
